pylxd/tests/models/test_container.py

```diff
@@ -514,6 +526,35 @@ def capture(request, context):
         with self.assertRaises(ValueError):
             res = self.container.files.put('/tmp/putted', data, mode=object)
 
+    def test_recursive_put(self):
+        """ Recursive put directory to the container """
```

Comment:
This test doesn't check that the header is actually set for directory when creating a directory.  Something also the lines of https://github.com/lxc/pylxd/blob/master/pylxd/tests/models/test_container.py#L471 where the headers are 'captured' during the test and can be checked would be very useful.